### ISSUE TITLE
Two transport 'location' headers cause an incorrect url

### ISSUE THREAD

**Issue by evgenykovalyov** [kennethreitz/requests#4037]
I noticed a bug while trying to simulate a login process for a service. At some point I receive a 302 response where there are two Location headers.

```
HTTP/1.1 302 Found
Location: https://some_url.com/
Accept-Ranges: bytes
Location: https://some_url.com/
```

While using the ```Session``` object, I noticed that both Location values are merged into one value like this:
```python
{'Location':'https://some_url.com/, https://some_url.com/'}
```

I added a temp split solution into ```SessionRedirectMixin.get_redirect_target()```, but I was wondering if you could come up with a permanent fix for this please?

```python
if resp.is_redirect:
            location = resp.headers['location']
            location = location.split(', ')[0]
```

Thanks in advance.

**Comment by sigmavirus24** [comment]
So I'm rather certain this is an issue with urllib3's Header handler. It needs a whitelist of which headers are allowed to be repeated and which can be joined with a `,` like you found here.

**Comment by nateprewitt** [comment]
Hey @evgenykovalyov, thanks for taking the time to open this issue. This is actually a dupe of #2939 and should be solved in Requests 3.0.0 with #3417. You'll likely have to wait until that's released for a more permanent solution. That is unless @sigmavirus24 has something else in mind.

I'm going to close this as resolved for now. Thanks again!
